cfme/tests/intelligence/test_optimization.py

```diff
@@ -69,4 +71,51 @@ def test_delete_generated_report():
             2.
             3. Report runs value for the optimization report must decrease by 1.
     """
-    pass
+    optimization_report = appliance.collections.optimization_reports.instantiate(
+        menu_name="Offline VMs with Snapshot"
+    )
+    opt_saved_report = optimization_report.queue()
+
+    # Note the value of runs for later comparison
+    old_run = optimization_report.runs
+
+    # There is an inconsistency between the time format for Optimization Report and Intel Report,
+    # to overcome that we pick the details from the SavedReport table so that it is easy to
+    # instantiate a SavedReport using ReportsCollection.
+    run_at_time = opt_saved_report.run_at.split()[1]
+
+    if BZ(1769333).blocks:
+        # convert run_at time from 12-hr to 24hr format
+        run_at_time = datetime.strftime(
+            datetime.strptime(
+                f"{opt_saved_report.run_at.split()[1]} {opt_saved_report.run_at.split()[2]}",
+                "%I:%M:%S %p",
+            ),
+            "%H:%M:%S",
+        )
+
+    view = navigate_to(optimization_report.collections.saved_reports, "All")
+    # This asserts the report exists for SavedReports accordion.
+    # If the report did not exist, RowNotFound error would be raised here.
+    row = next(
+        view.table.rows(
+            run_at__contains=run_at_time, name__contains=opt_saved_report.parent.parent.menu_name
+        )
+    )
```

Comment:
I don't see where the assert is happening. Won't this error out with a `RowNotFound` error rather than an `AssertionError`?

Reply:
What I meant by the comment was, if the row is found, we can assert that it exists/is_displayed, else it would raise a RowNotFound error. But I guess wrapping the statement in try and except block would be a good idea.